src/main/java/ch/njol/skript/expressions/ExprCoordinate.java

```diff
@@ -33,90 +35,159 @@
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.util.Kleenean;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Coordinate")
-@Description("Represents a given coordinate of a location. ")
-@Examples({"player's y-coordinate is smaller than 40:",
-		"	message \"Watch out for lava!\""})
-@Since("1.4.3")
-public class ExprCoordinate extends SimplePropertyExpression<Location, Number> {
-	
+@Description("Represents a given coordinate of locations or quaternions.")
+@Examples({
+	"player's y-coordinate is smaller than 40:",
+		"\tmessage \"Watch out for lava!\""
+})
+@Since("1.4.3, INSERT VERSION (quaternions)")
+public class ExprCoordinate extends SimplePropertyExpression<Object, Number> {
```

Comment:
Maybe either add Vector support to this class, or move quaternions over to the vector XYZComponent expression, as i think component fits better than coordinate.